cmd/releasego/update-azure-linux.go

```diff
@@ -0,0 +1,249 @@
+	if err != nil {
+		return err
+	}
+
+	golangSpecFileContent, err = updateSpecFile(assets, golangSpecFileContent)
+	if err != nil {
+		return err
+	}
+
+	prevGoArchiveName, err := extractGoArchiveNameFromSpecFile(golangSpecFileContent)
+	if err != nil {
+		return err
+	}
+
+	// Validation (as described in previous response)
+	if assets.GoSrcURL == "" || assets.GoSrcHash == "" {
+		return fmt.Errorf("invalid or missing GoSrcURL or GoSrcHash in assets.json")
+	}
+
+	golangSignaturesFileContent, err := downloadFileFromRepo(ctx, client, "microsoft", "azurelinux", "3.0-dev", golangSignaturesFilepath)
+	if err != nil {
+		return err
+	}
+
+	golangSignaturesFileContent, err = updateSignatureFile(golangSignaturesFileContent, prevGoArchiveName, path.Base(assets.GoSrcURL), assets.GoSrcHash)
+	if err != nil {
+		return err
+	}
+
+	cgManifestContent, err := downloadFileFromRepo(ctx, client, "microsoft", "azurelinux", "3.0-dev", cgManifestFilepath)
+	if err != nil {
+		return err
+	}
+
+	cgManifestContent, err = updateCGManifest(assets, cgManifestContent)
+	if err != nil {
+		return err
+	}
+
+	_ = golangSignaturesFileContent
+	_ = cgManifestContent
+
+	return nil
+}
+
+func loadBuildAssets(assetFilePath string) (*buildassets.BuildAssets, error) {
+	assets := new(buildassets.BuildAssets)
+
+	if err := stringutil.ReadJSONFile(assetFilePath, &assets); err != nil {
+		return nil, fmt.Errorf("error loading build assets: %w", err)
+	}
+
+	return assets, nil
+}
+
+func downloadFileFromRepo(ctx context.Context, client *github.Client, owner, repo, branch, filePath string) ([]byte, error) {
+	fileContent, err := githubutil.DownloadFile(ctx, client, owner, repo, branch, filePath)
+	if err != nil {
+		if errors.Is(err, githubutil.ErrNotExists) {
+			// Handle the specific case of the file not existing
+			return nil, fmt.Errorf("file '%s' not found in repository '%s' on branch '%s'", filePath, repo, branch)
+		} else {
+			// Handle other errors (network issues, authentication, etc.)
+			return nil, fmt.Errorf("failed to download file '%s': %w", filePath, err)
+		}
+	}
+
+	if len(fileContent) == 0 {
+		return nil, fmt.Errorf("downloaded file %s is empty", filePath)
+	}
+	return fileContent, nil
+}
+
+const (
+	golangSignaturesFilepath = "SPECS/golang/golang.signatures.json"
+	golangSpecFilepath       = "SPECS/golang/golang.spec"
+	cgManifestFilepath       = "cgmanifest.json"
+)
+
+var (
+	specFileGoFilenameRegex = regexp.MustCompile(`(%global ms_go_filename  )(.+)`)
+	specFileRevisionRegex   = regexp.MustCompile(`(%global ms_go_revision  )(.+)`)
+)
+
+func extractGoArchiveNameFromSpecFile(specContent []byte) (string, error) {
+	matches := specFileGoFilenameRegex.FindStringSubmatch(string(specContent))
+
+	if len(matches) < 2 {
+		return "", fmt.Errorf("no Go archive filename declaration found in spec content")
+	}
+
+	return strings.TrimSpace(matches[2]), nil
```

Comment:
It's odd seeing `len(matches) < 2 { return }` then `matches[2]`... I think it works based on what I know about `FindStringSubmatch` (only returns `nil` or a fully filled out slice) but it catches my eye because it wouldn't be correct if the slice length actually varied.

I think checking for `nil` is normal with this API because it explicitly says it will return `nil`, not an empty slice or something like that. (I see this being done in golang/go, for one.)